Astro runtime 11.0 (#3618)
astronomer/docs: astro/runtime-release-notes.md

Reviewer: > We now support runtime slime image with 11.0.0. Lets document that

I do not think so ...I dont think  we need to publicize that...

cc @jedcunningham

- New data-aware scheduling lets you use conditional logic (AND / OR) to schedule DAGs.
- You can now create your own labels for dynamically mapped tasks with templates, which makes it easier to search through mapped task instances.
- External XCom backends can now be configured to use object storage.
- Delivered several significant improvements to the Airflow UI. For example, you can now filter, view, and create datasets through the Airflow UI.
- New Listener API methods are considered stable and suitable for use in production.
- Added the ability to automatically pause a DAG after a pre-defined number of sequentially failed runs.
- Dataset URIs are validated when you enter them, and must conform to the rules set in AIP-60. See the [Dataset documentation](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/datasets.html) for more information.

For more information about the major changes in this release, see the [Airflow Blog](https://airflow.apache.org/blog/airflow-2.9.0/) or the [Airflow release notes](https://airflow.apache.org/docs/apache-airflow/2.9.0/release_notes.html#airflow-2-9-0-2024-04-08).

#### Upgrade to Python 3.12

> Astro Runtime now supports Python 3.12. However, [Pendulum](https://pendulum.eustace.io/) 2 does not support Python 3.12. If you upgrade to Python 3.12 and want to use Airflow, you also need to upgrade to Pendulum 3.

Author: ```suggestion
Airflow now supports Python 3.12. However, [Pendulum](https://pendulum.eustace.io/) 2 does not support Python 3.12. If you upgrade to Python 3.12 and want to use Airflow, you also need to upgrade to Pendulum 3.
```

> Astro Runtime now supports Python 3.12. However, [Pendulum](https://pendulum.eustace.io/) 2 does not support Python 3.12. If you upgrade to Python 3.12 and want to use Airflow, you also need to upgrade to Pendulum 3.

Reviewer 2: ```suggestion
Airflow now supports Python 3.12. However, [Pendulum](https://pendulum.eustace.io/) 2 does not support Python 3.12. If you upgrade to Python 3.12 and want to use Airflow, you also need to upgrade to Pendulum 3.
```
Now that this is nested, we can use the original wording

Refer to the [Airflow release notes](https://airflow.apache.org/docs/apache-airflow/2.9.0/release_notes.html#official-support-for-python-3-12-38025) for more information about any limitations in Python 3.12 support.

> #### Airflow bug fixes

Reviewer 2: ```suggestion
#### Bug fixes
```
Ok to cut since it's nested in the overall Airflow note

> - Fix a bug where after a task failed, and no longer exists in a DAG, you can now still access details about the DAG in the Grid View of the Airflow UI.([#38511](https://github.com/apache/airflow/pull/38511))

Reviewer 2: ```suggestion
- Fix a bug where after a task failed, and no longer exists in a DAG, you can now still access details about the DAG in the Grid View of the Airflow UI.
```

> - Fix a bug where Airflow would show `failed_upstream` when a dynamically mapped task was `skipped`.

Reviewer 2: ```suggestion
- Fixed a bug where Airflow would show `failed_upstream` when a dynamically mapped task was `skipped`.
```

:::

## Astro Runtime 11.0.0

- Release date: April 8, 2024
- Airflow version: 2.9.0

### Airflow 2.9.0

Astro Runtime 11.0.0 includes same-day support for Apache Airflow 2.9, which includes a number of new features and improvements. Airflow 2.9 includes the following changes:

- New data-aware scheduling functionality includes using conditional logic (AND and OR) or combining both dataset dependencies and time-based schedules.
- You can now create your own labels for dynamically mapped tasks with templates, instead of identifying them by searching through index numbers.
- UI improvements that include filtering, viewing, and creating Datasets through the UI, the ability to localize `dag_id` and `task_id` names, task log grouping, and improvements to DAG-level views such as calendar, run duration, and the audit log.

> 

Reviewer: ```suggestion
- AIP-58: Add object storage backend for xcom
```

For more information about the major changes in this release, see the [Airflow Blog](https://airflow.apache.org/blog/airflow-2.9.0/) or the [Airflow release notes](https://airflow.apache.org/docs/apache-airflow/2.9.0/release_notes.html#airflow-2-9-0-2024-04-08).

### Upgrade to Python 3.12

Airflow and the Astro Runtime now support Python 3.12. However, Pendulum 2 does not support Python 3.12. If you upgrade to Python 3.12 and want to use Airflow, you also need to upgrade to Pendulum 3.

Refer to the [Airflow release notes](https://airflow.apache.org/docs/apache-airflow/2.9.0/release_notes.html#official-support-for-python-3-12-38025) for more information about any limitations in Python 3.12 support.

### Additional improvements

> - New Listener API methods are considered stable and suitable for use in production. <!--https://github.com/apache/airflow/pull/36376-->

Reviewer 3: Should these be actual links, or is this just for context for editors?

Author: just context for editors! I'll remove them once we've got consensus

> - Added the ability to automatically set a DAG to `off` after a pre-defined number of sequentially failed runs. <!--https://github.com/apache/airflow/pull/36935-->

Reviewer 3: My 2c, "paused" is probably a better word to use than "off".

> - Added the ability to automatically set a DAG to `off` after a pre-defined number of sequentially failed runs. <!--https://github.com/apache/airflow/pull/36935-->

Author: ```suggestion
- Added the ability to automatically pause a DAG after a pre-defined number of sequentially failed runs.
```

:::

## Astro Runtime 11.0.0

- Release date: April 8, 2024
- Airflow version: 2.9.0

### Airflow 2.9.0

Astro Runtime 11.0.0 includes same-day support for Apache Airflow 2.9, which includes a number of new features and improvements. Airflow 2.9 includes the following changes:

- New data-aware scheduling functionality includes using conditional logic (AND and OR) or combining both dataset dependencies and time-based schedules.
- You can now create your own labels for dynamically mapped tasks with templates, instead of identifying them by searching through index numbers.
- Added the ability to store XComs in an object storage supported backend.

> - UI improvements that include filtering, viewing, and creating Datasets through the UI, the ability to localize `dag_id` and `task_id` names, task log grouping, and improvements to DAG-level views such as calendar, run duration, and the audit log.

Author: ```suggestion
- Delivered several significant improvements to the Airflow UI. For example, you can now filter, view, and create datasets through the Airflow UI..
```
